worker/src/ocall_bridge/ffi/get_quote.rs

```diff
@@ -0,0 +1,92 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+    Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use crate::ocall_bridge::bridge_api::{Bridge, RemoteAttestationOCall};
+use log::*;
+use sgx_types::{sgx_quote_nonce_t, sgx_quote_sign_type_t, sgx_report_t, sgx_spid_t, sgx_status_t};
+use std::slice;
+use std::sync::Arc;
+
+#[no_mangle]
+pub extern "C" fn ocall_get_quote(
+    p_sigrl: *const u8,
+    sigrl_len: u32,
+    p_report: *const sgx_report_t,
+    quote_type: sgx_quote_sign_type_t,
+    p_spid: *const sgx_spid_t,
+    p_nonce: *const sgx_quote_nonce_t,
+    p_qe_report: *mut sgx_report_t,
+    p_quote: *mut u8,
+    maxlen: u32,
+    p_quote_len: *mut u32,
+) -> sgx_status_t {
+    get_quote(
+        p_sigrl,
+        sigrl_len,
+        p_report,
+        quote_type,
+        p_spid,
+        p_nonce,
+        p_qe_report,
+        p_quote,
+        maxlen,
+        p_quote_len,
+        Bridge::get_ra_api(),
```

Comment:
This code would need to be derived with a procedural macro to red rid of the `dyn`.

Reply:
'Procedural macro' is not something I'm familiar with yet, I'll need to do some reading there.. thanks for the hint!

Reply:
It's has great potential, but I have not yet implemented one myself. :)